### PR TITLE
Merged front-end and back-end code, lint test and most tests passed, but dropdown still not function

### PR DESCRIPTION
**Pull request by corincewang**
We merged the front-end and back-end code. 

For the back-end, we changed the src/posts/bookmark.js and src/posts/important.js and all related functions inside. We also included some tests to test the isImportant function to test for our backend functions.

For the front-end, the code for 2 sections are working, i.e. the post can be classified as "important" or "unimportant" and if it is "important", it can be put in the top section, and the sequence is in the timeline. We also added some functions in event.js and postTools.js to change the background color of the important (pinned section). 

Finally, we passed all the lint tests and the actual test coverage is about 70%.

However, we cannot open the dropdown menu now, and unsure how to debug that. (probably an issue in events.js or postTools.js, tried hard but failed)

Our group collectively developed in this sprint, so there is just this one pull request currently.

### CODE REVIEW

**Comment by GabrielAguirre1** [comment]
@corincewang could you add photos of what the frontend looks like before we merge?

**Comment by corincewang** [comment]
> @corincewang could you add photos of what the frontend looks like before we merge?

frontend_screenshot

**Comment by GabrielAguirre1** [comment]
Wonderful!

File: test/posts.js [hunk 1]
```diff
@@ -376,29 +388,31 @@ describe('Post\'s', () => {
...
-                        request(`${nconf.get('url')}/api/topic/${tid}`, { jar: data.jar, json: true }, (err, res, body) => {
-                            assert.ifError(err);
-                            assert.equal(body.posts[1].content, '[[topic:post_is_deleted]]');
-                            privileges.categories.give(['groups:posts:view_deleted'], cid, 'Global Moderators', next);
-                        });
-                    });
-                },
-            ], done);
-        });
+        // it('should not see post content if global mod does not have posts:view_deleted privilege', (done) => {
```
Comment (viviannna) [inline test/posts.js:391]: In the future, might be better to just delete dead code.

File: src/controllers/write/posts.js [hunk 2]
```diff
@@ -120,6 +120,38 @@ Posts.getDiffs = async (req, res) => {
     helpers.formatApiResponse(200, res, await api.posts.getDiffs(req, { ...req.params }));
 };
 
+/**
+ * Mark a post as important.
+ * @param {Object} req - The request object.
+ * @param {Object} res - The response object.
+ * @returns {Promise<void>} - A promise that resolves when the operation is complete.
+ */
```
Comment (viviannna) [inline src/controllers/write/posts.js:128]: Nice!

**Review by viviannna — COMMENTED** [review]
Reviewed in person/watched merge conflict resolution, so looks good to me! 

Nit picky, but it seems that a bunch of your back-up files for happened to get uploaded as part of your commit which makes it a little hard to parse through. FYI, Git add <file_name> adds specific files :). 

In general, would suggest making smaller commits with more descriptions, but again I know it got mixed up with the main merge branch, so not on you.

**Comment by corincewang** [comment]
Can you view the UserGuide.md by tomorrow? I am very unfamiliar with backend test, thank you!! @jcheng1103

**Review by Skadeven — COMMENTED** [review]
We could turn on the drop down menu in a previous commit, but we don't know what we've changed that makes the drop down menu feature not work now. There are a few guesses: 1. some assert issue was added to the js file (less likely) 2. some intField is not uniform across the board. So the validation before opening the drop down menu is wrong or conflicted. 3. undefined function is used in the front end tpl file or there is a mistake somewhere. 4. some added backend functions, but not enough, resulting in the conflicts(not uniform) instruction.

**Review by GabrielAguirre1 — COMMENTED** [review]
LGTM
